File: app/eclipse/futurehome/src/no/futurehome/futurehome_app/widget/models/ShortcutM.java
package no.futurehome.futurehome_app.widget.models;

import android.content.Context;
import kotlin.Metadata;
import kotlin.NoWhenBranchMatchedException;
import kotlin.jvm.internal.Intrinsics;

@Metadata(d1 = {"\000.\n\002\030\002\n\002\020\000\n\000\n\002\020\b\n\000\n\002\020\016\n\002\b\b\n\002\030\002\n\000\n\002\030\002\n\002\b\004\n\002\020\013\n\002\b\004\b\b\030\0002\0020\001B\031\022\006\020\002\032\0020\003\022\b\020\004\032\004\030\0010\005¢\006\004\b\006\020\007J\026\020\f\032\0020\0052\006\020\r\032\0020\0162\006\020\017\032\0020\020J\t\020\021\032\0020\003HÆ\003J\013\020\022\032\004\030\0010\005HÆ\003J\037\020\023\032\0020\0002\b\b\002\020\002\032\0020\0032\n\b\002\020\004\032\004\030\0010\005HÆ\001J\023\020\024\032\0020\0252\b\020\026\032\004\030\0010\001HÖ\003J\t\020\027\032\0020\003HÖ\001J\t\020\030\032\0020\005HÖ\001R\021\020\002\032\0020\003¢\006\b\n\000\032\004\b\b\020\tR\023\020\004\032\004\030\0010\005¢\006\b\n\000\032\004\b\n\020\013¨\006\031"}, d2 = {"Lno/futurehome/futurehome_app/widget/models/ShortcutM;", "", "id", "", "name", "", "<init>", "(ILjava/lang/String;)V", "getId", "()I", "getName", "()Ljava/lang/String;", "getNameForLanguage", "context", "Landroid/content/Context;", "language", "Lno/futurehome/futurehome_app/widget/models/Language;", "component1", "component2", "copy", "equals", "", "other", "hashCode", "toString", "app_release"}, k = 1, mv = {2, 1, 0}, xi = 48)
public final class ShortcutM {
  private final int id;
  
  private final String name;
  
  public ShortcutM(int paramInt, String paramString) {
    this.id = paramInt;
    this.name = paramString;
  }
  
  public final int component1() {
    return this.id;
  }
  
  public final String component2() {
    return this.name;
  }
  
  public final ShortcutM copy(int paramInt, String paramString) {
    return new ShortcutM(paramInt, paramString);
  }
  
  public boolean equals(Object paramObject) {
    if (this == paramObject)
      return true; 
    if (!(paramObject instanceof ShortcutM))
      return false; 
    paramObject = paramObject;
    return (this.id != ((ShortcutM)paramObject).id) ? false : (!!Intrinsics.areEqual(this.name, ((ShortcutM)paramObject).name));
  }
  
  public final int getId() {
    return this.id;
  }
  
  public final String getName() {
    return this.name;
  }
  
  public final String getNameForLanguage(Context paramContext, Language paramLanguage) {
    String str1;
    Intrinsics.checkNotNullParameter(paramContext, "context");
    Intrinsics.checkNotNullParameter(paramLanguage, "language");
    String str2 = this.name;
    if (str2 != null)
      return str2; 
    int i = WhenMappings.$EnumSwitchMapping$0[paramLanguage.ordinal()];
    if (i != 1) {
      if (i != 2) {
        if (i != 3) {
          if (i != 4) {
            if (i == 5) {
              String str = paramContext.getString(2131689625);
              i = this.id;
              StringBuilder stringBuilder = new StringBuilder();
              stringBuilder.append(str);
              stringBuilder.append(" ");
              stringBuilder.append(i);
              str1 = stringBuilder.toString();
            } else {
              throw new NoWhenBranchMatchedException();
            } 
          } else {
            String str = str1.getString(2131689540);
            i = this.id;
            StringBuilder stringBuilder = new StringBuilder();
            stringBuilder.append(str);
            stringBuilder.append(" ");
            stringBuilder.append(i);
            str1 = stringBuilder.toString();
          } 
        } else {
          str1 = str1.getString(2131689604);
          i = this.id;
          StringBuilder stringBuilder = new StringBuilder();
          stringBuilder.append(str1);
          stringBuilder.append(" ");
          stringBuilder.append(i);
          str1 = stringBuilder.toString();
        } 
      } else {
        String str = str1.getString(2131689578);
        i = this.id;
        StringBuilder stringBuilder = new StringBuilder();
        stringBuilder.append(str);
        stringBuilder.append(" ");
        stringBuilder.append(i);
        str1 = stringBuilder.toString();
      } 
    } else {
      str1 = str1.getString(2131689557);
      i = this.id;
      StringBuilder stringBuilder = new StringBuilder();
      stringBuilder.append(str1);
      stringBuilder.append(" ");
      stringBuilder.append(i);
      str1 = stringBuilder.toString();
    } 
    return str1;
  }
  
  public int hashCode() {
    int i;
    int j = this.id;
    String str = this.name;
    if (str == null) {
      i = 0;
    } else {
      i = str.hashCode();
    } 
    return j * 31 + i;
  }
  
  public String toString() {
    int i = this.id;
    String str = this.name;
    StringBuilder stringBuilder = new StringBuilder("ShortcutM(id=");
    stringBuilder.append(i);
    stringBuilder.append(", name=");
    stringBuilder.append(str);
    stringBuilder.append(")");
    return stringBuilder.toString();
  }
}

File: app/futurehome-kotlin/no/futurehome/futurehome_app/widget/models/ShortcutM.kt
package no.futurehome.futurehome_app.widget.models

import android.content.Context

public data class ShortcutM(id: Int, name: String?) {
   public final val id: Int
   public final val name: String?

   init {
      this.id = var1;
      this.name = var2;
   }

   public operator fun component1(): Int {
      return this.id;
   }

   public operator fun component2(): String? {
      return this.name;
   }

   public fun copy(id: Int = var0.id, name: String? = var0.name): ShortcutM {
      return new ShortcutM(var1, var2);
   }

   public override operator fun equals(other: Any?): Boolean {
      if (this === var1) {
         return true;
      } else if (var1 !is ShortcutM) {
         return false;
      } else {
         var1 = var1;
         if (this.id != var1.id) {
            return false;
         } else {
            return this.name == var1.name;
         }
      }
   }

   public fun getNameForLanguage(context: Context, language: Language): String {
      if (this.name != null) {
         return this.name;
      } else {
         var var3: Int = ShortcutM.WhenMappings.$EnumSwitchMapping$0[var2.ordinal()];
         val var6: java.lang.String;
         if (var3 != 1) {
            if (var3 != 2) {
               if (var3 != 3) {
                  if (var3 != 4) {
                     if (var3 != 5) {
                        throw new NoWhenBranchMatchedException();
                     }

                     val var11: java.lang.String = var1.getString(2131689625);
                     var3 = this.id;
                     val var5: StringBuilder = new StringBuilder();
                     var5.append(var11);
                     var5.append(" ");
                     var5.append(var3);
                     var6 = var5.toString();
                  } else {
                     val var12: java.lang.String = var1.getString(2131689540);
                     var3 = this.id;
                     val var7: StringBuilder = new StringBuilder();
                     var7.append(var12);
                     var7.append(" ");
                     var7.append(var3);
                     var6 = var7.toString();
                  }
               } else {
                  val var8: java.lang.String = var1.getString(2131689604);
                  var3 = this.id;
                  val var13: StringBuilder = new StringBuilder();
                  var13.append(var8);
                  var13.append(" ");
                  var13.append(var3);
                  var6 = var13.toString();
               }
            } else {
               val var14: java.lang.String = var1.getString(2131689578);
               var3 = this.id;
               val var9: StringBuilder = new StringBuilder();
               var9.append(var14);
               var9.append(" ");
               var9.append(var3);
               var6 = var9.toString();
            }
         } else {
            val var10: java.lang.String = var1.getString(2131689557);
            var3 = this.id;
            val var15: StringBuilder = new StringBuilder();
            var15.append(var10);
            var15.append(" ");
            var15.append(var3);
            var6 = var15.toString();
         }

         return var6;
      }
   }

   public override fun hashCode(): Int {
      val var1: Int;
      if (this.name == null) {
         var1 = 0;
      } else {
         var1 = this.name.hashCode();
      }

      return this.id * 31 + var1;
   }

   public override fun toString(): String {
      val var1: Int = this.id;
      val var3: java.lang.String = this.name;
      val var2: StringBuilder = new StringBuilder("ShortcutM(id=");
      var2.append(var1);
      var2.append(", name=");
      var2.append(var3);
      var2.append(")");
      return var2.toString();
   }
}
